koiki/client.py

```diff
@@ -14,27 +14,33 @@
 class Client():
 
     def __init__(self, order, auth_token=koiki.auth_token, logger=koiki.logger):
-        self.order = order
+        self.order = Order(order)
         self.auth_token = auth_token
         self.host = koiki.host
         self.logger = logger
 
     def create_delivery(self):
         endpoint_req = CreateDelivery(self.order)
         req_body = self._authentication(endpoint_req)
-
         url = self._url(endpoint_req)
         self.logger.info('Koiki request to {}. body={}'.format(url, req_body))
 
         response = requests.post(url, json=req_body)
         response_body = json.loads(response.text)
 
-        if self._is_errored(response_body):
-            self._log(response.status_code, response.text, level='error')
-            return Error(response_body)
-        else:
-            self._log(response.status_code, self._masked_body(response_body))
-            return Delivery(response_body['envios'][0])
+        deliveries = []
+        for num, delivery_data in enumerate(response_body['envios']):
+            vendor = self.order.vendors[num]
+            delivery_data['order_id'] = self.order.order_id
+            delivery = Delivery(delivery_data, vendor)
+            if delivery._is_errored():
+                self._log(delivery.data.get('respuesta'), delivery.data, level='error')
+            else:
+                self._log(delivery.data.get('respuesta'), self._masked_body(delivery_data))
+
+            deliveries.append(delivery)
+
+        return deliveries
```

Comment:
I don't have an answer yet but I wasn't convinced of the responsibilities of this class since we created it. It seems to me it has too many. We should keep an eye on it.

Reply:
It will change when we refactore in the next PR's to conver part of that logic into Django model.

Reply:
Would you mind creating an issue so we don't forget? Also, it'll bring visibility to the entire team.